creator-node/src/routes/files.js

```diff
@@ -665,194 +519,35 @@ const getDirCID = async (req, res) => {
-      parentStoragePath,
-      { logContext: req.logContext },
-      filename
-    )
-    // Attempt to stream file to client.
     req.logger.info(`Retrieving ${storagePath} directly from filesystem`)
     return await streamFromFileSystem(req, res, storagePath)
   } catch (e) {
     req.logger.info(`Failed to retrieve ${storagePath} from FS`)
-
-    // ugly nested try/catch but don't want findCIDInNetwork to stop execution of the rest of the route
-    try {
-      // CID is the file CID, parse it from the storagePath
-      const CID = storagePath.split('/').slice(-1).join('')
-      const libs = req.app.get('audiusLibs')
-      await findCIDInNetwork(storagePath, CID, req.logger, libs)
-      return await streamFromFileSystem(req, res, storagePath)
-    } catch (e) {
-      req.logger.error(
-        `Error calling findCIDInNetwork for path ${storagePath}`,
-        e
-      )
-    }
   }
 
+  // Attempt to find and stream CID from other content nodes in the network
   try {
-    // For files not found on disk, attempt to stream from IPFS
-    // Cat 1 byte of CID in ipfs to determine if file exists
-    // If the request takes under 500ms, stream the file from ipfs
-    // else if the request takes over 500ms, throw an error
-    await ipfsSingleByteCat(ipfsPath, req.logContext, 500)
-
-    await new Promise((resolve, reject) => {
-      req.app
-        .get('ipfsAPI')
-        .catReadableStream(ipfsPath)
-        .on('data', (streamData) => {
-          res.write(streamData)
-        })
-        .on('end', () => {
-          res.end()
-          resolve()
-        })
-        .on('error', (e) => {
-          reject(e)
-        })
-    })
+    // CID is the file CID, parse it from the storagePath
+    const CID = storagePath.split('/').slice(-1).join('')
+    const libs = req.app.get('audiusLibs')
+    await findCIDInNetwork(storagePath, CID, req.logger, libs)
+    return await streamFromFileSystem(req, res, storagePath)
   } catch (e) {
+    req.logger.error(
+      `Error calling findCIDInNetwork for path ${storagePath}`,
+      e
+    )
     // Unset the cache-control header so that a bad response is not cached
     res.removeHeader('cache-control')
     return sendResponse(req, res, errorResponseServerError(e.message))
   }
 }
 
-/**
- * Perform IPFS verification with retries
- *
- * Use retries because for some reason IPFS (very rarely) fails due to some non-deterministic error. Seems to be with the verification step; the files are correct.
- * Wait fixed timeout between each retry, hopefully addresses IPFS non-deterministic errors
- * @param {Object} req
- * @param {File[]} resizeResp resizeImage.js response; should be a File[] of resized images
- * @param {string} dirCID the directory CID from `resizeResp`
- * @param {number?} maxRetries the max number of retries for ipfs verification
- * @param {boolean?} enableIPFSAdd flag to enable or disable ipfs daemon add
- */
-const _dirCIDIPFSVerificationWithRetries = async function (
```

Comment:
i think we might want to keep this. what this (badly named) function is doing is checking that using the IPFS deterministic content hashing algorithm (not the actual daemon) that the file contents are equal to the expected hash

Reply:
there's an optional enableIPFSAdd param which actually determines if it adds to the IPFS daemon or just uses the algorithm

Reply:
Good call -- I see that now that I looked into it more. I addressed that and removed more IPFS stuff in another commit